gammapy/irf/effective_area_table.py

```diff
@@ -360,8 +364,24 @@ class EffectiveAreaTable2D(object):
     >>> from gammapy.datasets import load_aeff2D_fits_table
     >>> aeff2D = EffectiveAreaTable2D.from_fits(load_aeff2D_fits_table())
     >>> offset = Angle(0.6, 'degree')
-    >>> energies = Quantity(np.logspace(0, 1, 60), 'TeV')
-    >>> eff_area = aeff2D.evaluate(offset, energies)
+    >>> energy = Quantity(np.logspace(0, 1, 60), 'TeV')
+    >>> eff_area = aeff2D.evaluate(offset, energy)
+
+    Create ARF fits file for a given offest and energy binning:
+
+    >>> import numpy as np
+    >>> from astropy.coordinates import Angle
+    >>> from astropy.units import Quantity
+    >>> from gammapy.irf import EffectiveAreaTable2D
+    >>> from gammapy.datasets import load_aeff2D_fits_table
+    >>> aeff2D = EffectiveAreaTable2D.from_fits(load_aeff2D_fits_table())
+    >>> offset = Angle(0.43, 'degree')
+    >>> nbins = 50
+    >>> energy = Quantity(np.logspace(0, 1, nbins+1), 'TeV')
```

Comment:
@kingj90 Sounds good to me, please proceed like this here, or even clean up other code in Gammapy in this PR or a separate one to follow this style.

Although, I haven't thought about it much / looked at which parts or the code would change in which way, so I reserve the right to say something else in the future. One thing that *might* make sense in the future is an `EnergyBinning` class that is a wrapper or sub-class of the n+1 length quantity corresponding the the `EBOUNDS` extension, with a way to read / write this to a HDU and convenience attributes or functions to get e.g. the array of lower or upper energy bin edges. I only vaguely remember, but I think in the HESS software we have this as an `EnergyAxis` class (although I'm not sure "axis" is the best term for this).